src/lib/modernExtend.ts

```diff
@@ -750,18 +750,117 @@ export function co2(args?: Partial<NumericArgs>) {
-        description: 'Remaining battery in %',
-        unit: '%',
-        scale: 2,
-        access: 'STATE_GET',
-        ...args,
-    });
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const payload: KeyValueAny = {};
+            if (msg.data.hasOwnProperty('batteryPercentageRemaining') && (msg.data['batteryPercentageRemaining'] < 255)) {
+                // Some devices do not comply to the ZCL and report a
+                // batteryPercentageRemaining of 100 when the battery is full (should be 200).
+                const dontDividePercentage = model.meta && model.meta.battery && model.meta.battery.dontDividePercentage;
+                let percentage = msg.data['batteryPercentageRemaining'];
+                percentage = dontDividePercentage ? percentage : percentage / 2;
+                if (args.percentage) payload.battery = precisionRound(percentage, 2);
+            }
+
+            if (msg.data.hasOwnProperty('batteryVoltage') && (msg.data['batteryVoltage'] < 255)) {
+                // Deprecated: voltage is = mV now but should be V
+                if (args.voltage) payload.voltage = msg.data['batteryVoltage'] * 100;
+
+                if (model.meta && model.meta.battery && model.meta.battery.voltageToPercentage) {
+                    payload.battery = batteryVoltageToPercentage(payload.voltage, model.meta.battery.voltageToPercentage);
+                }
+            }
+
+            if (msg.data.hasOwnProperty('batteryAlarmState')) {
+                const battery1Low = (
+                    msg.data.batteryAlarmState & 1<<0 ||
+                    msg.data.batteryAlarmState & 1<<1 ||
+                    msg.data.batteryAlarmState & 1<<2 ||
+                    msg.data.batteryAlarmState & 1<<3
+                ) > 0;
+                const battery2Low = (
+                    msg.data.batteryAlarmState & 1<<10 ||
+                    msg.data.batteryAlarmState & 1<<11 ||
+                    msg.data.batteryAlarmState & 1<<12 ||
+                    msg.data.batteryAlarmState & 1<<13
+                ) > 0;
+                const battery3Low = (
+                    msg.data.batteryAlarmState & 1<<20 ||
+                    msg.data.batteryAlarmState & 1<<21 ||
+                    msg.data.batteryAlarmState & 1<<22 ||
+                    msg.data.batteryAlarmState & 1<<23
+                ) > 0;
+                if (args.lowStatus) payload.battery_low = battery1Low || battery2Low || battery3Low;
+            }
+
+            return payload;
+        },
+    }];
+
+    const defaultReporting: ReportingConfigWithoutAttribute = {min: '1_HOUR', max: 'MAX', change: 10};
+
+    const configure: Configure = async (device, coordinatorEndpoint, logger) => {
+        if (args.percentageReporting) {
+            if (args.percentageReportingConfig) {
+                await setupAttributes(device, coordinatorEndpoint, 'genPowerCfg', [
```

Comment:
Please use:

```js
await setupAttributes(device, coordinatorEndpoint, 'genPowerCfg', [
                    {attribute: 'batteryPercentageRemaining', ...(args.percentageReportingConfig ?? defaultReporting)},
                ], logger);
```

Same for voltage reporting below.

Reply:
Done.